test/main.cpp

```diff
@@ -458,12 +458,14 @@ TEST_CASE("Ordering", "[test]")
     SECTION("reverse Base")
     {
         auto comp = tree.value_comp();
-
+        bool actually_checked = false;
         REQUIRE(std::is_sorted(tree.rbegin(), tree.rend(),
         [&](const value_type& a, const value_type& b)
         {
+            actually_checked = true;
             return !comp(a, b);
         }));
+        REQUIRE(actually_checked);
```

Comment:
As a demonstration of my explanation of why this test used to work, this test used to fail on the previous version code